### PR TITLE
fix: refactor Button storybook

### PR DESCRIPTION
**Pull request by burakukula**
Refactor Button to the new story format and add the overview of component variants for testing purposes

## PR Checklist

- [ ] I have read the relevant `readme.md` file(s)
- [ ] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [ ] Tests are added/updated/not required
- [ ] Tests are passing
- [ ] Storybook stories are added/updated/not required
- [ ] Usage notes are added/updated/not required
- [ ] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [ ] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/forma-36-react-components/src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -1,47 +1,79 @@
...
+      small
+    </Button>
+    <Button buttonType="primary">default size</Button>
+    <Button buttonType="primary" size="large">
+      large
+    </Button>
+  </>
+);
+
+export const statesOverview = (args: ButtonProps) => (
```
Comment (burakukula) [inline packages/forma-36-react-components/src/components/Button/Button.stories.tsx:59]: Or maybe we would prefer to have just one story with the overview of all the variants and states?

Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx:59]: Don't many stories work best with Chromatic?

Comment (burakukula) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx:59]: Maybe let's try to merge this one and I will create PR with some changes so we would be able to see it in Chromatic? WDYT?

**Review by m10l — CHANGES_REQUESTED** [review]
I think splitting up the states to show more variants would be better and help us catch any issues

So something like this:
- isActive - primary, positive, warning, negative, muted, naked
- disabled - primary, positive, warning, negative, muted, naked
- indicateDropdown - primary, positive, warning, negative, muted, naked
- loading - primary, positive, warning, negative, muted, naked

File: packages/forma-36-react-components/src/components/Button/Button.stories.tsx (hunk 1 above)
Comment (m10l) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx:59]: You can already see this in Chromatic here: https://www.chromatic.com/pullrequest?appId=5fd1dda724cc620021ace8c5&number=721&view=changes (click details on UI Review GitHub check).

I think having a single story would be nicer for Storybook users. It shouldn't matter too much as Chromatic will still pick up everything, I just guess the visual diffs might be a bit noisier.

Comment (burakukula) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx]: Sorry, didn't see your comment. Ok, sure, I will merge them into one story.

**Comment by burakukula** [comment]
@m10l ✅ please have a look, if you have a moment. 🙂

**Review by m10l — APPROVED** [review]

File: packages/forma-36-react-components/src/components/Button/Button.stories.tsx (hunk 1 above)
Comment (denkristoffer) [reply packages/forma-36-react-components/src/components/Button/Button.stories.tsx]: I think I'm misunderstanding something, I only see a single button in the Basic story, is that right? How can Chromatic detect changes in the other types then?

**Pull request merged**